### ISSUE TITLE
How does Classification work for person re-ID?

### ISSUE THREAD

**Issue by Greyvend** [KaiyangZhou/deep-person-reid#61]
It's more of a generic question, but still. You have this `train_imgreid_xent.py` file for "train image model with cross entropy loss". I've checked the code and it's not clear, how is it possible to use classification in person re-identification? We don't know the number of classes (persons) in advance, so how many classes should the CNN have? Or is it just 2 classes - "this person"/"some other person"?

**Comment by MuadDev** [comment]
In general person reid goes somewhat like this:
- During training there are N classes, and the network must predict the correct class for each of the identities in the training set.
- During inference the output of the network is taken from a layer **before** the last prediction layer. This output is then used as a descriptor for the input image and its distance to other images' descriptors determines if it belongs to a similar person or not.

**Comment by Greyvend** [comment]
@MuadDev thanks for the explanation. According to the experiments, is it clear, is Triplet loss always better than this approach?

**Comment by yang502** [comment]
model = models.init_model(name=args.arch, num_classes=dataset.num_train_pids, loss={'xent'}, use_gpu=use_gpu) 
when we define the training model, we set the num_classes which computed in the train dataset.

**Comment by northeastsquare** [comment]
@ciwei123  cosin distance or l2 distance
